inc/compatibilities/woocommerce.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/**
+ * Class Optml_elementor_builder
+ *
+ * @reason Adding selectors for background lazyload
+ */
+class Optml_woocommerce extends Optml_compatibility {
+
+
+	/**
+	 * Should we load the integration logic.
+	 *
+	 * @return bool Should we load.
+	 */
+	function should_load() {
+		include_once( ABSPATH . 'wp-admin/includes/plugin.php' );
+		return Optml_Main::instance()->admin->settings->use_lazyload() && is_plugin_active( 'woocommerce/woocommerce.php' );
+	}
+
+	/**
+	 * Register integration details.
+	 */
+	public function register() {
+		add_filter( 'optml_should_ignore_image_tags', [$this, 'check_product_page'], 10, 1 );
+	}
+	/**
+	 * Add ignore page lazyload flag.
+	 *
+	 * @param bool $old_value Previous returned value.
+	 *
+	 * @return bool If we should lazyload the page.
+	 */
+	public function check_product_page( $old_value ) {
+		if ( is_product() ) {
```

Comment:
This fix looks a bit too loose, can't we exclude exactly the problematic area? Also, we can investigate if there is any custom lazyload that we could do to gain maximum efficiency when using Optimole

Reply:
The problem is on every single product page ( that function is not so intuitive in the name, but it does exactly this check that we need https://docs.woocommerce.com/document/conditional-tags/ ) and because there is not too much to lazyload on the page with just the product it seems a good idea to exclude the page. If we want to keep the lazyload on this pages as they do not include lazyload for the related products for example so it will still gain from ours we could add their lazyload src `data-src` to our flag list when we detect their plugin and this should also give our benefits without interfering with their lazyload ?